homeassistant/components/shelly/__init__.py

```diff
@@ -270,7 +273,10 @@ async def _async_setup_rpc_entry(hass: HomeAssistant, entry: ShellyConfigEntry)
                 async_create_issue_unsupported_firmware(hass, entry)
                 await device.shutdown()
                 raise ConfigEntryNotReady
-        except (DeviceConnectionError, MacAddressMismatchError) as err:
+            runtime_data.script_events = await get_rpc_scripts_event_types(
```

Comment:
```suggestion
            runtime_data.rpc_script_events = await get_rpc_scripts_event_types(
```
To be clear that it is only relevant for RPC devices